ecs-cli/modules/cli/compose/entity/service/service_helper.go

```diff
@@ -15,22 +15,76 @@ package service
 
 import (
 	"fmt"
+	"sort"
+	"time"
 
 	log "github.com/Sirupsen/logrus"
-	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/utils/waiters"
+	ecscli "github.com/aws/amazon-ecs-cli/ecs-cli/modules/commands"
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
 )
 
+// UpdateServiceTimeout is the time that the CLI will wait to check if the
+// count of running tasks is changing. If it has not changed then an error is thrown
+// after UpdateServiceTimeout minutes
+const DefaultUpdateServiceTimeout = 5
+const latestEventWindow = 5
+
+// serviceEvents is a wrapper for []*ecs.ServiceEvent
+// that allows us to sort it by the time stamp
+type serviceEvents []*ecs.ServiceEvent
+
+func (s serviceEvents) Len() int {
+	return len(s)
+}
+func (s serviceEvents) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+func (s serviceEvents) Less(i, j int) bool {
+	time1 := *s[i].CreatedAt
+	time2 := *s[j].CreatedAt
+	return time1.Before(time2)
+}
+
+// logNewServiceEvents logs events that have not been logged yet
+func logNewServiceEvents(loggedEvents map[string]bool, events []*ecs.ServiceEvent, cmdTimestamp time.Time) {
+
+	// sort the events so that newer ones are printed last
+	sort.Sort(serviceEvents(events))
+	for _, event := range events {
+		if _, ok := loggedEvents[*event.Id]; !ok {
+			// New event that has not been logged yet
+			loggedEvents[*event.Id] = true
```

Comment:
No. `cmdTimeStamp` is when the command (`compose service...`) was run. So this will log all events since the command has been run. There is no bug here. I have tested this thoroughly.